### PR TITLE
Add annotations to support building Third Party Classes

### PR DESCRIPTION
**Pull request by bduong**
Add new annotations and processors in order to support creating builders for pojos that aren't within the source tree.

This allows builders to be created for class that aren't owned by the developer.

### CODE REVIEW

File: buildable/src/main/java/buildable/config/BuildField.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Documented
+@Retention(RetentionPolicy.SOURCE)
+public @interface BuildField {
```
Comment (brad-neighbors) [inline buildable/src/main/java/buildable/config/BuildField.java]: The other annotations have pretty verbose javadoc, which I quite like (really the only kind of javadoc that's worth doing, IMO), we should be consistent.

File: buildable/src/main/java/buildable/config/BuildableSpec.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Documented
+@Target({ElementType.TYPE})
+@Retention(RetentionPolicy.SOURCE)
+public @interface BuildableSpec {
```
Comment (brad-neighbors) [inline buildable/src/main/java/buildable/config/BuildableSpec.java]: The other annotations have pretty verbose javadoc, which I quite like (really the only kind of javadoc that's worth doing, IMO), we should be consistent.

File: buildable/src/main/java/buildable/config/InjectBuildable.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Documented
+@Target({ElementType.FIELD})
+@Retention(RetentionPolicy.SOURCE)
+public @interface InjectBuildable {
```
Comment (brad-neighbors) [inline buildable/src/main/java/buildable/config/InjectBuildable.java]: javadoc

File: buildable/src/main/java/buildable/config/processor/ClassWriterUtil.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    /**
+     *
+     * @param builder
+     * @param buildableField
+     * @param fieldName
+     * @param builtWith
+     * @param fieldClassName
+     * @param builderName
+     */
+    static void buildFieldMethod(TypeSpec.Builder builder, VariableElement buildableField, String fieldName,
```
Comment (brad-neighbors) [inline buildable/src/main/java/buildable/config/processor/ClassWriterUtil.java]: Should this just be a private method in your new processor?

File: buildable/src/main/java/buildable/annotation/processor/BuildableCombinedProcessor.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+import static javax.tools.Diagnostic.Kind.NOTE;
+
+/**
+ * An annotation processor that will process both @Buildable and @BuildableSpec annotated classes.
+ *
+ * This processor also shares the classes that will have builders created for between the two annotation processors. This allows builder methods
+ * that takes other builders to utilize builders created by the other annotation processor.
+ */
+@SupportedSourceVersion(SourceVersion.RELEASE_8)
+public class BuildableCombinedProcessor  extends AbstractProcessor {
```
Comment (brad-neighbors) [inline buildable/src/main/java/buildable/annotation/processor/BuildableCombinedProcessor.java]: I think I'd just simplify this to BuildableProcessor

**Comment by bduong** [comment]
Added annotations for building using the constructor

**Review by brad-neighbors — APPROVED** [review]
Looks good Ben.  Neat.

**Pull request merged**